### PR TITLE
Doc: extend application sphinx extension to support flash arguments and fix uses

### LINKED ISSUE #64130: Documentation: zephyr-app-commands where goals are more than one word are badly rendered
**Describe the bug**
In the documentation, there is a few places where the source rst tries to define a `zephyr-app-commands` with a multiword "goal", but the rendered version is split into one line per word
See for example:
https://docs.zephyrproject.org/latest/samples/boards/nrf/nrf53_sync_rtc/README.html#building-the-application-for-nrf5340dk-nrf5340-cpuapp
https://docs.zephyrproject.org/latest/boards/arm/adafruit_itsybitsy_m4_express/doc/index.html#debugging 
Source:
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/boards/nrf/nrf53_sync_rtc/README.rst?plain=1#L31
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/adafruit_itsybitsy_m4_express/doc/index.rst?plain=1#L182

**Expected behavior**
The command in the documentation is correct

**Impact**
Confused readers

**Logs and console output**
See above

### PR DESCRIPTION
**Pull request by aescolar**
Passing arguments to west flash is needed in many cases but there was no way to properly generate documentation with such examples. With this new "flash-args" option we can now.

And fix the west flash command examples which needed it using this new flash-args option.

Fixes #64130

---

And a bonus fix:

    doc: samples net mqtt_sn_publisher fix west invokation

    Fix the west invokation. "-t" is added automatically
    to the run goal, otherwise we get a "-t -t" command.

### CODE REVIEW

**Comment by kartben** [comment]
Nice, thank you!! Will review more closely after 3.5 is out but it looks good to me.

File: samples/net/mqtt_sn_publisher/README.rst [hunk 1]
```diff
@@ -68,7 +68,7 @@ Then, locate your zephyr directory and type:
 .. zephyr-app-commands::
    :zephyr-app: samples/net/mqtt_sn_publisher
    :board: native_posix_64
-   :goals: build -t run
+   :goals: build run
```
[the remark this answers is not in the document]
Comment (kartben) [reply samples/net/mqtt_sn_publisher/README.rst:71]: I think the challenge is that if there are multiple goals, there's then no way to tell which set of args belong to which goal though?

[the remark this answers is not in the document]
Comment (gmarull) [reply samples/net/mqtt_sn_publisher/README.rst:71]: right, so then it make sense. is then this change correct? `west run` doesn't exist

[the remark this answers is not in the document]
Comment (aescolar) [reply samples/net/mqtt_sn_publisher/README.rst:71]: goals is a list of goals, so then we'd need to know to which goal "goal-args" applies.

[the remark this answers is not in the document]
Comment (aescolar) [reply samples/net/mqtt_sn_publisher/README.rst:71]: when a goal does not exist, the extension code adds the `-t` on its own

File: doc/_extensions/zephyr/application.py [hunk 2]
```diff
@@ -92,6 +92,9 @@ class ZephyrAppCommandsDirective(Directive):
     \:west-args:
```
Comment (jukkar) [inline doc/_extensions/zephyr/application.py:92]: Typo in commit `doc: extend application sphinx extension to support flash arguments` text
s/flah-args/flash-args/

File: samples/net/mqtt_sn_publisher/README.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (gmarull) [reply samples/net/mqtt_sn_publisher/README.rst:71]: seems like a mishandling of goals, then.

File: doc/_extensions/zephyr/application.py (hunk 2 above)
Comment (aescolar) [reply doc/_extensions/zephyr/application.py:92]: fixed :)

**Review by carlescufi — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**